src/main/kotlin/eu/europa/ec/eudi/openid4vci/internal/http/AuthorizationEndpointClient.kt

```diff
@@ -241,7 +242,7 @@ internal class AuthorizationEndpointClient(
         credentialConfigurationId: CredentialConfigurationIdentifier,
     ): AuthorizationDetail =
         with(NimbusAuthorizationDetail.Builder(AuthorizationType(OPENID_CREDENTIAL))) {
-            if (credentialIssuerId.toString() != authorizationIssuer.toString()) {
+            if (credentialIssuerId.toString() != authorizationIssuer) {
```

Comment:
Please add a new private function 

```kotlin
private fun isCredentialIssuerAuthorizationServer(): Boolean = 
    credentialIssuerId.toString() == authorizationIssuer
```
This method evaluates whether the Credential Issuer is 
- a resource server protected by an external authorization server ( representing by `false`), or whether 
- both an authorization & resource server (represented by `true`)

Replace the line 244 to use the new function

```kotlin 
if (!isCredentialIssuerAuthorizationServer) {
   // Add location
}
```